realm/realm-library/src/androidTest/java/io/realm/RealmSchemaTests.java

```diff
@@ -25,38 +31,47 @@
 import io.realm.entities.AllJavaTypes;
 import io.realm.entities.Owner;
 import io.realm.internal.Util;
+import io.realm.rule.TestRealmConfigurationFactory;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+@RunWith(AndroidJUnit4.class)
+public class RealmSchemaTests {
 
-public class RealmSchemaTests extends AndroidTestCase {
+    @Rule
+    public final TestRealmConfigurationFactory configFactory = new TestRealmConfigurationFactory();
 
-    public static final String CLASS_ALL_JAVA_TYPES = "AllJavaTypes";
     private DynamicRealm realm;
     private RealmSchema realmSchema;
 
-    @Override
-    protected void setUp() throws Exception {
-        super.setUp();
-        RealmConfiguration realmConfig = new RealmConfiguration.Builder(getContext())
+    @Before
+    public void setUp() {
+        RealmConfiguration realmConfig = configFactory.createConfigurationBuilder()
                 .schema(AllJavaTypes.class, Owner.class)
                 .build();
-        Realm.deleteRealm(realmConfig);
         Realm.getInstance(realmConfig).close(); // create Schema
-        this.realm = DynamicRealm.getInstance(realmConfig);
+        realm = DynamicRealm.getInstance(realmConfig);
         realmSchema = this.realm.getSchema();
-        this.realm.beginTransaction();
+        realm.beginTransaction();
     }
 
-    @Override
-    protected void tearDown() throws Exception {
-        super.tearDown();
+    @After
+    public void tearDown() {
         realm.cancelTransaction();
         realm.close();
     }
 
-    public void testGetAllClasses() {
+    @Test
+    public void getAll() {
         Set<RealmObjectSchema> objectSchemas = realmSchema.getAll();
         assertEquals(5, objectSchemas.size());
 
-        List<String> expectedTables = Arrays.asList(CLASS_ALL_JAVA_TYPES, "Owner", "Cat", "Dog", "DogPrimaryKey");
+        List<String> expectedTables = Arrays.asList(AllJavaTypes.CLASS_NAME, "Owner", "Cat", "Dog", "DogPrimaryKey");
```

Comment:
`objectSchemas ` does not seem to contain the classes other than "AllJavaTypes" and "Owner";
How about adding the assertion that checks all expected classes are contained like:

```
        List<String> expectedTables = new ArrayList<String>(Arrays.asList(AllJavaTypes.CLASS_NAME, "Owner"));
        for (RealmObjectSchema objectSchema : objectSchemas) {
            if (!expectedTables.remove(objectSchema.getClassName())) {
                fail(objectSchema.getClassName() + " is unexpected.");
            }
        }
        assertTrue(expectedTables.toString() + " was not found.", expectedTables.isEmpty());
```